js/ui/codeView.js

```diff
@@ -1261,7 +1269,39 @@ var CodeViewManager = new Lang.Class({
     },
 
     _addSession: function(actor) {
-        this._sessions.push(new CodingSession({ app: actor }));
+        const session = new CodingSession({ app: actor });
+
+        // When the app is minimized the WM doesn't emit the destroy signal if
+        // the app is closed, for this reason we need to listen to the 'destroy'
+        // signal of the actor if it's minimized, to be able to remove from the
+        // CodingSession list and avoid possible gnome-shell crash
+        let destroyAppHandlerId = 0;
+        let destroyToolboxHandlerId = 0;
+        session.connect('minimized', (session) => {
+            if (session.app) {
+                destroyAppHandlerId = session.app.connect('destroy',
+                    this.handleDestroyWindow.bind(this));
+            }
+
+            if (session.toolbox) {
+                destroyToolboxHandlerId = session.toolbox.connect('destroy',
+                    this.handleDestroyWindow.bind(this));
+            }
+        });
+        session.connect('unminimized', (session) => {
+            if (destroyAppHandlerId) {
+                if (session.app)
```

Comment:
I wonder if this check is actually required, since `destroyAppHandlerId` only is set if `session.app` is set in the other callback. (Same below.)